src/Raven.Server/Rachis/Follower.cs

```diff
@@ -687,13 +698,31 @@ private void MaybeNotifyLeaderThatWeAreStillAlive(TransactionOperationContext co
                 midpointTerm = _engine.GetTermForKnownExisting(context, midpointIndex);
             }
 
-
-            while (minIndex < maxIndex)
+            while ((midpointTerm == negotiation.PrevLogTerm && midpointIndex == negotiation.PrevLogIndex) == false)
             {
                 _engine.Timeout.Defer(_connection.Source);
 
                 _engine.ValidateTerm(_term);
-                
+
+                if (midpointIndex == negotiation.PrevLogIndex && midpointTerm != negotiation.PrevLogTerm)
+                {
+                    // our appended entries has been diverged, same index with different terms.
+                    var msg = $"Our appended entries has been diverged, same index with different terms. " +
+                              $"My index/term {midpointIndex}/{midpointTerm}, while yours is {negotiation.PrevLogIndex}/{negotiation.PrevLogTerm}.";
+                    if (_engine.Log.IsInfoEnabled)
+                    {
+                        _engine.Log.Info($"{ToString()}: {msg}");
+                    }
+                    connection.Send(context, new LogLengthNegotiationResponse
+                    {
+                        Status = LogLengthNegotiationResponse.ResponseStatus.Acceptable,
```

Comment:
Why is this acceptable here?

Reply:
This is acceptable with `LastLogIndex` = 0, meaning that the leader should send us full snapshot.